skorch/tests/test_net.py

```diff
@@ -571,10 +571,13 @@ def test_callback_same_inferred_name_twice(self, net_cls, module_cls):
         cb1.__class__.__name__ = 'some-name'
         net = net_cls(module_cls, callbacks=[cb0, cb1])
 
-        with pytest.raises(ValueError) as exc:
-            net.initialize()
-        expected = "The callback name 'some-name' appears more than once."
-        assert str(exc.value) == expected
+        net.initialize()
+
+        cbs = dict(net.callbacks_)
+        assert 'some-name_1' in cbs
```

Comment:
Unrelated to this PR, but shouldn't dashes in the custom names be replaced with underscores?

Reply:
Why?

Reply:
Names with dashes can not be used to construct keyword arguments, so `net.set_params(callbacks__some-name_1__foo=10)` is not possible. Or is the assumption that if the user has chosen a name with a dash, then he does not plan to use it with `set_params()`?

Reply:
> Or is the assumption that if the user has chosen a name with a dash, then he does not plan to use it with set_params()?

In general, when a user wants to use commodities like `set_params`, they should make sure that their parts cooperate. We can only give them guidance.

For your example, I believe the dash is still no problem:

`net.set_params(**{'callbacks__some-name_1__foo':10})`